vpr/src/place/move_transactions.cpp

```diff
@@ -40,7 +49,7 @@ void apply_move_blocks(const t_pl_blocks_to_be_moved& blocks_affected) {
 
     //Swap the blocks, but don't swap the nets or update place_ctx.grid_blocks
     //yet since we don't know whether the swap will be accepted
-    for (int iblk = 0; iblk < blocks_affected.num_moved_blocks; ++iblk) {
+    for (size_t iblk = 0; iblk < blocks_affected.moved_blocks.size(); ++iblk) {
```

Comment:
To be safe, could do num_moved = blocks_affected.moved_blocks.size() before the loop -- it is slightly safer from a compiler optimization point of view, as otherwise we are counting on the compiler to figure out .size() doesn't need to be called each time.